alacritty/src/display/hint.rs

```diff
@@ -150,7 +150,9 @@ impl HintState {
             let bounds = self.matches[index].clone();
             let action = hint.action.clone();
 
-            self.stop();
+            if !hint.persist {
+                self.stop();
+            }
 
             // Hyperlinks take precedence over regex matches.
             let hyperlink = term.grid()[*bounds.start()].hyperlink();
```

Comment:
```suggestion
            // Exit hint mode unless it requires explicit dismissal.
            if !hint.persist {
                self.stop();
            }
            
            let bounds = self.matches[index].clone();
            let action = hint.action.clone();

            // Hyperlinks take precedence over regex matches.
            let hyperlink = term.grid()[*bounds.start()].hyperlink();
            
```

The order of operations here is a bit confusing, not really related to your PR but I'd fix it while we're at it. This way the data aggregation for the return value is separate from the modification to the state itself.

The little comment on the conditional will also hopefully make things more clear.

Reply:
I'm trynna see how I can fix this after reorganizing the operations. Still a little rusty at Rust, sry! :P

```rs
error[E0502]: cannot borrow `*self` as mutable because it is also borrowed as immutable
   --> alacritty/src/display/hint.rs:154:17
    |
142 |         let hint = self.hint.as_ref()?;
    |                    ------------------ immutable borrow occurs here
...
154 |                 self.stop();
    |                 ^^^^^^^^^^^ mutable borrow occurs here
...
158 |             let action = hint.action.clone();
    |                          ------------------- immutable borrow later used here

For more information about this error, try `rustc --explain E0502`.
```

`stop` needs a mutable reference to `self`, but we're creating an immutable reference above.

Reply:
Ah, I see what the issue is. Calling `self.stop()` clears `self.hint` and `self.matches` ([ref](https://github.com/alacritty/alacritty/blob/7257c49313984f88c2f6b550a81ce838c75deda2/alacritty/src/display/hint.rs#L63-L68)). So, trying to access `self.matches[index].clone()` would cause a panic because of an index out of bounds. I think the order of the operations must be kept as is.. unless there's a different way to do this that I'm missing?